### PR TITLE
Config reading fix and more robust curve plot flag handling

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
More robust parsing of acas configs
Utility functions for filtering points
More robust handling of missing flags

## Related Issue
ACAS-273

## How Has This Been Tested?
Verified curve fit still works as expected
Verified that curve curator (including curve rendering) still works as expected

### CODE REVIEW

File: R/rendering.R [hunk 1]
```diff
@@ -336,9 +336,79 @@ getCurveIDAnalsysiGroupResults <- function(curveids, ...) {
...
 #' 
+filterFlaggedPoints <- function(points, returnGood = TRUE) {
+  badFlag <- "knocked out"
+  if(returnGood) {
+    return(subset(points, userFlagStatus!=badFlag & preprocessFlagStatus!=badFlag & algorithmFlagStatus!=badFlag & tempFlagStatus!=badFlag))
+  } else {
+    return(subset(points, userFlagStatus==badFlag | preprocessFlagStatus==badFlag | algorithmFlagStatus==badFlag | tempFlagStatus==badFlag))
+  }
+}
+applyParsedParametersToFitData <- function(fitData, parsedParams, protocolDisplayValues) {
```
Comment (bffrost) [inline R/rendering.R:357]: Could you add a comment here as to what this new function does?

File: R/rendering.R [hunk 2]
```diff
@@ -336,9 +336,79 @@ getCurveIDAnalsysiGroupResults <- function(curveids, ...) {
...
+        colorCategories <- uniqueKeys[, color:=rep(plotColors, length.out = .N)][ , name:=get(key)]
+        fitData <- merge(fitData, colorCategories, by = key)
+    }
 
-plotCurve <- function(curveData, params, outFile = NA, ymin = NA, logDose = FALSE, logResponse = FALSE, ymax = NA, xmin = NA, xmax = NA, height = 300, width = 300, showGrid = FALSE, showLegend = FALSE, showAxes = TRUE, drawCurve = TRUE, drawFlagged = FALSE, plotMeans = FALSE, drawStdDevs = FALSE, addShapes = FALSE, labelAxes = FALSE, curveXrn = c(NA, NA), mostRecentCurveColor = NA, axes = c("x","y"), modZero = TRUE, drawPointsForRejectedCurve = racas::applicationSettings$server.curveRender.drawPointsForRejectedCurve, plotColors = c("black"),curveLwd = 1, plotPoints = TRUE, xlabel = NA, ylabel = NA) {
+    if("category" %in% names(fitData)) {
+      fitData <- fitData[exists("category") & (!is.null(category) & category %in% c("inactive","potent")), c("fittedMax", "fittedMin") := {
+        responseMean <- mean(points[[1]][userFlagStatus!="knocked out" & preprocessFlagStatus!="knocked out" & algorithmFlagStatus!="knocked out" & tempFlagStatus!="knocked out",]$response)
```
Comment (bffrost) [inline R/rendering.R]: Some of this is really dense and it's hard for me to follow what this is doing, and why it's doing it. This function appears to have a bunch of important logic about built-in default behaviors. I'd appreciate a lot more comments throughout this new function to describe those behaviors.

File: R/rendering.R (hunk 1 above)
Comment (brianbolt) [reply R/rendering.R:357]: I added a function doc string here.

**Comment by brianbolt** [comment]
Tried to add a lot of comments and things here but this code was not commented well to begin with so it's hard to dig into a lot of aspects and provide commenting now. Opening this for re-review now.

**Review by bffrost — APPROVED** [review]
LGTM

**Pull request merged**